pkg/controllers/provisioningv2/rke2/machinedrain/machinedrain.go

```diff
@@ -72,21 +50,45 @@ func (h *handler) OnChange(_ string, secret *corev1.Secret) (*corev1.Secret, err
 		return secret, err
 	}
 
+	oldSecret := secret.DeepCopy()
+	defer func() {
+		if secret == nil || secret.Annotations == nil {
```

Comment:
The closure doesn't keep the state at the time it was created. So, if one of the functions below returns a `nil` secret, then it will be `nil` when this `defer` runs.

Reply:
Makes sense